### PR TITLE
fix(43154): Syntax error in emitted output from `tsc` when exporting value of symbol key

### LINKED ISSUE #43154: Syntax error in emitted output from `tsc` when exporting value of symbol key
# Bug Report

### 🔎 Search Terms
syntax error tsc declaration class symbol  TS1003 Identifier expected

### 🕗 Version & Regression Information
Every version from 3.3.3 - 4.2.3 and nightly

- This is a crash

### ⏯ Playground Link
https://www.typescriptlang.org/play?ts=4.3.0-dev.20210308#code/MYewdgzgLgBAyjAvPAngWwEYgDYAoCUA3AFDDYCGEEMAYiCDAN7EyszTlQCWwMA2nAC6BRgF9i4gKYAPAA4gATrFCRYaSVAAWIACZJa9AYMJA

### 💻 Code
```ts
const S = Symbol();
class Foo {
    static [S](){}
}
export const method = Foo[S];
```

### 🙁 Actual behavior
Output of the `d.ts` follows:
```ts
declare const S: unique symbol;
declare class Foo {
    static [S](): void;
}
export declare const method: typeof Foo.[S];
export {};
```

There is a syntax error on line 5: `typeof Foo.[S]`

### 🙂 Expected behavior
A dot should not be emitted when the key is a symbol

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #43154

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -4798,7 +4798,8 @@ namespace ts {
                     }
                     function shouldWriteTypeOfFunctionSymbol() {
                         const isStaticMethodSymbol = !!(symbol.flags & SymbolFlags.Method) &&  // typeof static method
-                            some(symbol.declarations, declaration => hasSyntacticModifier(declaration, ModifierFlags.Static));
+                            some(symbol.declarations, declaration =>
+                                hasSyntacticModifier(declaration, ModifierFlags.Static) && isMethodDeclaration(declaration) && !isComputedPropertyName(declaration.name));
```
Comment (sandersn) [inline src/compiler/checker.ts:4802]: what happens with property declarations? does that have an existing test?

**Review by sandersn — CHANGES_REQUESTED** [review]
One question below, plus: does this fix the computed property name with a literal type instead of a symbol type? If yes, you should add a test for that too.

File: src/compiler/checker.ts (hunk 1 above)
Comment (weswigham) [reply src/compiler/checker.ts:4802]: Moreover, when the static member with a computed name that we're referencing has a `unique symbol` type itself, what do we do? I expect we'd need to emit `(typeof Foo)[key]`.

Comment (a-tarasyuk) [reply src/compiler/checker.ts:4802]: @sandersn I've added additional tests. 

@weswigham `(typeof Foo)[key]` is this allowed in declarations? Wouldn't that throw an error like
`"'key' refers to a value, but is being used as a type ...."`?

Comment (weswigham) [reply src/compiler/checker.ts:4802]: Well... `(typeof Foo)[typeof key]` :3

**Comment by a-tarasyuk** [comment]
.

**Comment by a-tarasyuk** [comment]
Reopen

**Comment by a-tarasyuk** [comment]
Reopen

**Comment by a-tarasyuk** [comment]
Reopen

**Pull request closed without merging**